### DISCUSSION TITLE
Debugging React Native

### DISCUSSION THREAD

**Discussion by Aleksandern** [mobxjs/mobx#2745] [General]
Hi guys,

Could somebody share your experience in debugging mobx state on react-native.
I use [mobx-remotedev](https://github.com/zalmoxisus/mobx-remotedev) + [react-native-debugger](https://github.com/jhen0409/react-native-debugger) .
But mobx-remotedev doesn't support MobX 6 and has some issues with async actions.

Is there any alternative?

Thanks.

**Comment by kubk** [comment]
mobx-remotedev does work for me with Mobx 6. I had to use a custom function to get rid of `remotedev` decorator:

```js
const spy = require('mobx-remotedev/lib/spy').default;

export const addToDevtools = (store: object) => {
  if (process.env.NODE_ENV !== 'production') {
    spy(store, { global: true, name: store.constructor.name });
  }
};

```

Usage:
```js
class Counter {
  value = 0;
  constructor() {
    makeAutoObservable(this);
    addToDevtools(this);
  }
}
```

But I agree Mobx lacks proper devtools support :cry:

**Comment by mweststrate** [comment]
IIRC reactotron does supports MobX / spy events as well
